### PR TITLE
Broker Stats

### PR DESCRIPTION
**Pull request by danadesrosiers**
Added some functionality to provide some basic insight into the state of the BrokerService.  Related to issue #39.

* Created two counters in the ReferenceWrapper to record the total number of messages received and delivered for each queue.
* Added endpoint to the BrokerService to allow external services to poll for Broker stats, including a list of all subscribers and the current totals for each queue.
* Added method to BrokerClient to consume the GetBrokerStats() endpoint.  Calls all partitions of the BrokerService and aggregates the results.
* Added method to BrokerClient to unsubscribe a service by queue name.

### CODE REVIEW

**Review by loekd — COMMENTED** [review]
How do you feel about abstracting the implementation behind an interface, e.g. `IStatsCollector`? You could inject an implementation that logs to ETW, or one that logs to memory, etc.

**Comment by danadesrosiers** [comment]
I think the current implementation would have to change a bit to make that possible, but I'm willing to do it if it adds value.  Can you explain a little more what you are hoping to get out of this?  Right now, it's just incrementing a counter, not logging anything.

**Comment by danadesrosiers** [comment]
I added some code.  Is that the kind of thing you're think of?  Maybe add a few more callbacks, like `OnSubscribe()` and `OnUnsubscribe()`, so users can implement an `IStatsCollector` to do whatever they need.  It could potentially replace some or all `ServiceEventSourceMessage()` calls.

File: ServiceFabric.PubSubActors/Stats/StatsCollector.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Threading.Tasks;
+using ServiceFabric.PubSubActors.State;
+
+namespace ServiceFabric.PubSubActors.Stats
+{
+    public class StatsCollector : IStatsCollector
+    {
+        public Task OnMessageReceived(ReferenceWrapper subscriber, MessageWrapper messageWrapper)
+        {
+            subscriber.TotalReceived++;
```
Comment (loekd) [inline ServiceFabric.PubSubActors/Stats/StatsCollector.cs:10]: This is exactly what I mean yes, except for the call to set the `TotalReceived` on the subscriber. I was thinking that this `StatsCollector` could be renamed into `InMemoryStatCollector` (which keeps track of delivered messages in memory) and that we could have alternate implementations, like one that sends a message into ETW and one that sends custom telemetry into AppInsights.

File: ServiceFabric.PubSubActors/Stats/StatsCollector.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            return Task.FromResult(true);
+        }
+
+        public Task OnMessageDelivered(ReferenceWrapper subscriber, MessageWrapper messageWrapper)
+        {
+            subscriber.TotalDelivered++;
+            return Task.FromResult(true);
+        }
```
Comment (loekd) [inline ServiceFabric.PubSubActors/Stats/StatsCollector.cs:18]: We could also add `OnMessageDeliveryFailed`

**Review by loekd — CHANGES_REQUESTED** [review]

**Comment by danadesrosiers** [comment]
Closing this pull request so we can delete the SoCreate fork.  New PR is #64.

**Pull request closed without merging**